doc/manual/src/architecture/store/reference-scanning.md

```diff
@@ -0,0 +1,57 @@
+# Reference scanning
+
+As discussed in a previous section, a store object is made up of file system objects and references to other store objects.
+
+Suppose we are trying to create a new store object that *ought* to have references, but we only have plain file system data.
+For example, imagine the file system data is a "shared library" (so, dll, dylib) that depends on other shared libraries, and those other shared libraries are already inside other store objects.
+How might we do this?
+
+One way would be to have the user manually write down this information.
+Another way would be to teach Nix to parse those files' headers and convert the "domain specific" dependency information within them.
+Both of these are a huge amount of work, both initially and as ongoing maintenance, and therefore not very attractive options.
+
+Nix instead opts to merely scan the file for store paths without making any attempt to parse it.
+This "simple, stupid" approach actually works quite well most of the time, and imposes very little burden on the end user.
```

Comment:
If you say most of the time, you should explain where it does not work. I assume this is [what you intended here](2dd490623fcfc237e63eb99ae70feb6cd1ddd7978df62b53f308bf3cb2e47114R24), but I think it should be there in place. I'm pretty sure Eelco has something very similar very clearly formulated in his thesis.

Reply:
Examples I have heard of off hand are compressed JAR files and compressed man pages. Are those in the manual?